### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by A-Russell23**
## Changes:

- A few minor fixes to the EBNF
- Added an explanation of our syntax with examples. I've marked which sections are in progress/not implemented yet. This should be updated as required.

### CODE REVIEW

File: C_Flat Definition.md [hunk 1]
```diff
@@ -152,16 +235,16 @@
...
-`<Function-Call>::= <Identifier> '(' *{<Identifier>} ')' ';'`
+`<Function-Call>::= <Identifier> '(' *{<Identifier> | <Logic-Statement> | <Expression>} ')' ';'`
 
 ### Loops:
 
 `<While-Loop>::= 'while' '(' <Logic-Statement> ')' <Block>`
 
 ### Keywords:
 
-`<Keyword>::= 'if' || 'else' || 'while' || 'var' || 'func' | 'true' | 'false`
```
Comment (Spartan-Bailey) [inline C_Flat Definition.md:163]: quotation on false keyword is not complete

File: C_Flat Definition.md [hunk 2]
```diff
@@ -139,11 +221,12 @@
...
 ### Variables:
 
-`<Declaration>::= 'var’ (<Identifier> | <Assignment>) ‘;’`
+`<Declaration>::= 'var’ (<Identifier> ';'| <Assignment>)`
 
 `<Identifier>::= <Word>`
 
 `<Word>::= 1*(a-zA-Z) - <Keywords>`
+
 `<Assignment>::= <Identifier> '=' (<Expression> | ' " '<Word>' " ' | <Logic-Statement>) ‘;’`
```
Comment (Spartan-Bailey) [inline C_Flat Definition.md:230]: Semi colon needs to be removed from Assignment as it is already covered in Declaration

Comment (Spartan-Bailey) [inline C_Flat Definition.md:230]: Assignment needs to be put on a new line

**Review by Spartan-Bailey — CHANGES_REQUESTED** [review]

File: C_Flat Definition.md (hunk 2 above)
Comment (A-Russell23) [reply C_Flat Definition.md:230]: There was an issue with duplicate semi-colons but I've fixed this here by adding the semi-colon after Identifier in Declaration, and keeping it as part of Assignment. So this should all be correct. Hopefully that makes sense!

Just to note, as I think you might have gotten confused here - anything highlighted in red is whats been removed and anything highlighted in green is whats been added/changed. :)

Comment (A-Russell23) [reply C_Flat Definition.md:230]: Yep thats been done here in this change :)

File: C_Flat Definition.md (hunk 1 above)
Comment (A-Russell23) [reply C_Flat Definition.md:163]: Thanks for letting me know, I'll fix this now :)

**Review by Spartan-Bailey — APPROVED** [review]

**Pull request merged**